iphone/Classes/TiUIWebView.m

```diff
@@ -672,21 +698,33 @@ -(CGFloat)contentWidthForWidth:(CGFloat)value
 }
 
 #pragma mark WebView Delegate
-
 - (BOOL)webView:(UIWebView *)webView shouldStartLoadWithRequest:(NSURLRequest *)request navigationType:(UIWebViewNavigationType)navigationType
 {
 	NSURL * newUrl = [request URL];
-
-	if ([self.proxy _hasListeners:@"beforeload"])
-	{
-		NSDictionary *event = newUrl == nil ? nil : [NSDictionary dictionaryWithObjectsAndKeys:[newUrl absoluteString], @"url", NUMINT(navigationType), @"navigationType", nil];
-		[self.proxy fireEvent:@"beforeload" withObject:event];
-	}
-
-	if (navigationType != UIWebViewNavigationTypeOther) {
-		RELEASE_TO_NIL(lastValidLoad);
-	}
-
+    
+    if ([self.proxy _hasListeners:@"beforeload"])
+    {
+        NSDictionary *event = newUrl == nil ? nil : [NSDictionary dictionaryWithObjectsAndKeys:[newUrl absoluteString], @"url", NUMINT(navigationType), @"navigationType", nil];
+        [self.proxy fireEvent:@"beforeload" withObject:event];
+    }
+    
+    if (navigationType != UIWebViewNavigationTypeOther) {
+        RELEASE_TO_NIL(lastValidLoad);
+    }
+    
+    if (blacklistedURLs) {
+        for (NSString*check in blacklistedURLs)
+        {
+            if ([[newUrl absoluteString] rangeOfString:check options:NSCaseInsensitiveSearch].location != NSNotFound) {
+                if([self.proxy _hasListeners:@"onStopBlacklistedUrl"]) {
+                    NSDictionary* evt = [NSDictionary dictionaryWithObjectsAndKeys:[TiUtils stringValue:request], @"url", @"Webview did not load blacklisted url.", @"messsage", nil];
```

Comment:
Oh that sorry, thought you was talking about the if statement. Yeah it works, i cant pass in the URL as it causes a weird bug which we disguised, where the page will finish loading before reaching the return statement.